contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -536,7 +540,7 @@ func (s *SchedulerServer) getDriver() (driver bindings.SchedulerDriver) {
 }
 
 func (s *SchedulerServer) Run(hks hyperkube.Interface, _ []string) error {
-	if n := len(s.mesosRoles); n == 0 || n > 2 || (n == 2 && s.mesosRoles[0] != "*" && s.mesosRoles[1] != "*") {
+	if n := len(s.frameworkRoles); n == 0 || n > 2 || (n == 2 && s.frameworkRoles[0] != "*" && s.frameworkRoles[1] != "*") {
```

Comment:
OK, after rethought I suggest to implement a subset validation only during bootstrap and agree with @jdef.

A superset is indeed legal but indeed very confusing when setting up the k8sm scheduler.